Symfony/CS/Console/Output/ProcessOutput.php

```diff
@@ -0,0 +1,91 @@
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Console\Output;
+
+use Symfony\Component\Console\Output\StreamOutput;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+use Symfony\CS\FixerFileProcessedEvent;
+
+/**
+ * Output writer to show the process of a FixCommand.
+ *
+ * @internal
+ */
+class ProcessOutput
+{
+    /**
+     * File statuses map.
+     *
+     * @var array
+     */
+    private static $eventStatusMap = array(
+        FixerFileProcessedEvent::STATUS_UNKNOWN => array('symbol' => '?', 'format' => '%s', 'description' => 'unknown'),
+        FixerFileProcessedEvent::STATUS_INVALID => array('symbol' => 'I', 'format' => '<bg=red>%s</bg=red>', 'description' => 'invalid file syntax, file ignored'),
+        FixerFileProcessedEvent::STATUS_SKIPPED => array('symbol' => 'S', 'format' => '<fg=cyan>%s</fg=cyan>', 'description' => 'Skipped'),
+        FixerFileProcessedEvent::STATUS_NO_CHANGES => array('symbol' => '.', 'format' => '%s', 'description' => 'no changes'),
+        FixerFileProcessedEvent::STATUS_FIXED => array('symbol' => 'F', 'format' => '<fg=green>%s</fg=green>', 'description' => 'fixed'),
+        FixerFileProcessedEvent::STATUS_EXCEPTION => array('symbol' => 'E', 'format' => '<bg=red>%s</bg=red>', 'description' => 'error'),
+        FixerFileProcessedEvent::STATUS_LINT => array('symbol' => 'E', 'format' => '<bg=red>%s</bg=red>', 'description' => 'error'),
+    );
+
+    /**
+     * Stream output instance.
+     *
+     * @var StreamOutput
+     */
+    private $output;
+
+    /**
+     * Event dispatcher instance.
+     *
+     * @var EventDispatcher
+     */
+    private $eventDispatcher;
+
+    /**
+     * @param EventDispatcher $dispatcher
+     * @param StreamOutput    $output
+     */
+    public function __construct(EventDispatcher $dispatcher, StreamOutput $output)
+    {
+        $this->output = $output;
+        $this->eventDispatcher = $dispatcher;
+        $this->eventDispatcher->addListener(FixerFileProcessedEvent::NAME, array($this, 'onFixerFileProcessed'));
+    }
+
+    public function onFixerFileProcessed(FixerFileProcessedEvent $event)
+    {
+        $status = self::$eventStatusMap[$event->getStatus()];
+        $this->output->write($this->output->isDecorated() ? sprintf($status['format'], $status['symbol']) : $status['symbol']);
+    }
+
+    public function printLegend()
+    {
+        $symbols = array();
+
+        foreach (self::$eventStatusMap as $status) {
+            $symbol = $status['symbol'];
+            if ('' === $symbol || isset($symbols[$symbol])) {
+                continue;
+            }
+
+            $symbols[$symbol] = sprintf('%s-%s', $this->output->isDecorated() ? sprintf($status['format'], $symbol) : $symbol, $status['description']);
+        }
+
+        $this->output->write(sprintf("\nLegend: %s\n", implode(', ', $symbols)));
```

Comment:
The start + end linebreak might be a break with the console style guide discussed at Symfony at the moment, should I change this?